### PR TITLE
Fix tramp alien index find file with relative paths.

### PR DESCRIPTION
**Pull request by TrunovS**
Alien indexing method returned absolute file paths which in turn failed `projectile-find-file` in Tramp session for me. Got that working with fix in the way `projectile-dir-files-alien` create list of files with relative paths instead of absolute paths. And I believe its solves the way `projectile-find-file-in-dir` works with alien indexing with tramp.

Closes: #1275

### CODE REVIEW

**Comment by TrunovS** [comment]
@bbatsov It seams something wrong with test concerning projectile-project-root and its cache? But I cant understand why.

File: projectile.el [hunk 1]
```diff
@@ -1194,12 +1194,18 @@ function is executing."
...
-    (cond
-    ((eq vcs 'git)
-     (nconc (projectile-files-via-ext-command directory (projectile-get-ext-command vcs))
-            (projectile-get-sub-projects-files directory vcs)))
-    (t (projectile-files-via-ext-command directory (projectile-get-ext-command vcs))))))
+  (let ((vcs (projectile-project-vcs directory))
+        ;; getting directory name without ":" for tramp connections
+        (c_directory (car (last (split-string directory ":")))))
+
+    (mapcar (lambda (f)
```
Comment (bbatsov) [inline projectile.el]: I haven't looked at the test failure, but I can tell you that walking over all the files and applying an extra transformation is going to have a terrible performance impact on any non-trivial project.

Comment (TrunovS) [reply projectile.el]: Thanks!) Ok, that's was kind of hack to make file list more compact without long prefix `/home/user/<proj_root>/` for hg repos. It's seams better place for that was `projectile-hg-command` (change hg locate to hg files). By chance `hg files` also returns project file list in more compact way without long prefix.

**Comment by TrunovS** [comment]
@bbatsov Last commit likely fixed some test failures for Travis emacs 26 and snaptshot, but not so fo emacs 25.

**Comment by TrunovS** [comment]
This feature is somewhat buggy and still need to be refined.

**Pull request closed without merging**